src/learning/lda.rs

```diff
@@ -0,0 +1,250 @@
+//! Latent Diriclhet Allocation Module
+//!
+//! Contains an implementation of Latent Diriclhet Allocation (LDA) using
+//! [Gibbs Sampling](http://psiexp.ss.uci.edu/research/papers/sciencetopics.pdf)
+//!
+//! LDA is typically used for textual analysis.  It assumes that a topic can be modeled
+//! as a distribution of words, and that a document can be modeled as a distribution of
+//! topics.  Thus, the output is these distributions on the input documents.
+//!
+//! Gibbs sampling is a Morkov Chain Monto Carlo algorithm that iteratively approximates
+//! the above distributions.
+//!
+//! This module doesn't use any training.  It uses unsupervised learning to estimate
```

Comment:
LDA (in its original form) is in a sort of strange place with respect to machine learning, in that it's designed to classify only the input data, and nothing more.  In this sense, the training and prediction stage are actually identical.  One could certainly use the distribution that was discovered to classify additional documents, but for whatever reason, that's not how it was used.  I'll have more to say about this in a later comment.